src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java

```diff
@@ -140,8 +149,10 @@ private static String getGeneratorFieldValue(FieldType field, Generator generato
             case RATED_S -> String.valueOf(generator.getRatedS());
             case COUNTRY,
                 NOMINAL_VOLTAGE,
-                VOLTAGE_LEVEL_ID -> getVoltageLevelFieldValue(field, generator.getTerminal().getVoltageLevel());
+                VOLTAGE_LEVEL_ID -> getVoltageLevelFieldValue(field, null, generator.getTerminal().getVoltageLevel());
             case CONNECTED -> getTerminalFieldValue(field, generator.getTerminal());
+            case FREE_PROPERTIES -> generator.getProperty(propertyName);
```

Comment:
not needed as we add a case in getFieldValue

Reply:
done